packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -276,45 +270,42 @@ function mapValueWithFallbacks(
 				return value;
 			}
 		}
-		default:
+		case "string":
+		// TODO:
+		// This should detect invalid strings. Something like @stdlib/regexp-utf16-unpaired-surrogate could be used to do this.
+		// See SchemaFactory.string for details.
+		case "boolean":
 			return value;
+		case "object": {
+			if (value === null || isFluidHandle(value)) {
+				return value;
+			}
+		}
+		default:
+			throw new TypeError(`Received unsupported leaf value: ${value}.`);
```

Comment:
invalid data now errors instead of being treated as valid. This seems much better since it might encounter things like big ints, objects where they arn't expected etc, and this function is supposed validate.